server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -172,6 +172,10 @@ public void run() {
 
     private void start(final long replicationId) {
         try (ReplicationRef<SegmentReplicationTarget> replicationRef = onGoingReplications.get(replicationId)) {
+            if (replicationRef == null) {
+                logger.trace("Cannot find any running replication with id [{}]", replicationId);
```

Comment:
I don't think a log line is necessary here. Just include a comment that this is for handling edge cases where the reference is removed before the `ReplicationRunner` is started by the threadpool.